### ISSUE TITLE
Latest Component Update Not Authenticating

### ISSUE THREAD

**Issue by isablend** [albertogeniola/meross-homeassistant#71]
Since updating to the latest component version, I'm getting an error message

Invalid config for [meross_cloud]: expected int for dictionary value @ data['meross_cloud']['username']. Got 'xxxxxxxx@gmail.com'. (See /config/configuration.yaml, line 273). Please check the docs at https://www.home-assistant.io/components/meross_cloud

I checked the userid and password and indeed they were out of sync, so I updated and verified using the IoS app that they right credentials were in operation.   So I can see all devices in the IoS app and they respond in the app, but the HA component doesn't authenticate.

I haven't changed the config between version of the component.   Any ideas?

**Comment by LaskaBozzolo** [comment]
I have the same problem here, after updating.
I cannot even restart home assistant because of that (I receive the error isablend reported), need full reboot.

**Comment by aferraz** [comment]
Hi, 
 same error for me. 
I have removed the lines from configuration.yaml, from Integration menu I removed Meross and installed again inserting username and password.
The error disappear but the component doesn't work anyway, now the errore is:
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 30

**Comment by Jonah1970** [comment]
I am having similar problems and was unable to get HA to restart to a point where the Front End would load.  An extract from my log is given below and this repeats indefinately:

2020-04-22 15:12:27 WARNING (MainThread) [homeassistant.helpers.entity] Update of switch.christmas_tree is taking over 10 seconds
2020-04-22 15:12:27 ERROR (SyncWorker_15) [custom_components.meross_cloud.common] Component version: 0.3.4.3-2, Message: "A command timeout occurred while handling cloud_io function."
Traceback (most recent call last):
  File "/config/custom_components/meross_cloud/common.py", line 83, in func_wrapper
    return func(*args, **kwargs)
  File "/config/custom_components/meross_cloud/switch.py", line 35, in update
    self._device.get_status(force_status_refresh=True)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 134, in get_status
    self._state = self._get_status_impl()
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 88, in _get_status_impl
    data = self.get_sys_data()['all']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 119, in get_sys_data
    return self.execute_command("GET", ALL, {}, online_check=False)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 294, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10
2020-04-22 15:12:28 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform meross_cloud is taking over 10 seconds.
2020-04-22 15:12:37 WARNING (MainThread) [homeassistant.helpers.entity] Update of switch.outside_lights is taking over 10 seconds
2020-04-22 15:12:37 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up meross_cloud platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/config/custom_components/meross_cloud/sensor.py", line 176, in async_setup_entry
    sensor_entities = await hass.async_add_executor_job(sync_logic)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/meross_cloud/sensor.py", line 164, in sync_logic
    elif plug.type.startswith("mss310") or plug.supports_consumption_reading():
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 152, in supports_consumption_reading
    return CONSUMPTIONX in self.get_abilities()
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 124, in get_abilities
    self._abilities = self.execute_command("GET", ABILITY, {})['ability']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 294, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10
2020-04-22 15:12:37 ERROR (SyncWorker_4) [custom_components.meross_cloud.common] Component version: 0.3.4.3-2, Message: "A command timeout occurred while handling cloud_io function."
Traceback (most recent call last):
  File "/config/custom_components/meross_cloud/common.py", line 83, in func_wrapper
    return func(*args, **kwargs)
  File "/config/custom_components/meross_cloud/switch.py", line 35, in update
    self._device.get_status(force_status_refresh=True)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 134, in get_status
    self._state = self._get_status_impl()
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 88, in _get_status_impl
    data = self.get_sys_data()['all']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 119, in get_sys_data
    return self.execute_command("GET", ALL, {}, online_check=False)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 294, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10
2020-04-22 15:12:47 WARNING (MainThread) [homeassistant.helpers.entity] Update of switch.christmas_tree is taking over 10 seconds
2020-04-22 15:12:47 ERROR (SyncWorker_18) [custom_components.meross_cloud.common] Component version: 0.3.4.3-2, Message: "A command timeout occurred while handling cloud_io function."
Traceback (most recent call last):

I was able to get HA back up back renaming the meross_cloud folder and restarting, now I just get the following error in the log but at least HA is working:

2020-04-22 15:36:32 ERROR (MainThread) [homeassistant.setup] Setup failed for meross_cloud: Unable to import component: No module named 'custom_components.meross_cloud'

**Comment by albertogeniola** [comment]
Hi @isablend, @LaskaBozzolo and @aferraz ,

I was able to reproduce the problem you mentioned. That occurs when the integration is used with the old **configuration.yaml** setting. Removing the entry and reinstalled the component would have fixed the problem.

Anyways, thanks to your indication, I was able to identify an error in the latest code (regression) which was causing the error you experienced. I've fixed it in the latest version, which is now available on HACS. 

@Jonah1970 : your case seems similar to the ones experienced by many users nowdays. Some new meross hardware (HW version 2.0 and FW version 2.1.4+) are incompatible with this component as Meross has changed something in theur code and that is preventing the library for working as expected. Please report [your case here](https://github.com/albertogeniola/meross-homeassistant/issues/67).

**Comment by aferraz** [comment]
@albertogeniola thanks, I can confirm that now it's working

**Comment by Jonah1970** [comment]
Hi @albertogeniola, just so you are aware ... I have 2 x MSS210 smart plugs at HW version 1.0.0 & FW version 1.12.  I have been using these successfully since installing HACS so I doubt this could be an incompatibility with new HW/FW version numbers.  Also, I am surprised that the issue stops HA from starting and wondered whether it ought to have come up in safe mode to allow the problem to be addressed.
For info I am running in Docker on a Synology.
Thanks

**Comment by albertogeniola** [comment]
Hi @Jonah1970 ,

This might have been solved by the latest BETA release. Do you mind testing it and report back the outcome of your test?

**Comment by Jonah1970** [comment]
Hi @albertogeniola, I installed version 20200424.1 which I believe is the latest Beta.  I can confirm that HA does restart however I am still getting multiple erros in the log (see below) and although the 2 switches appear in the Front End they are not operating.

2020-04-24 09:16:42 WARNING (MainThread) [homeassistant.helpers.entity] Update of switch.christmas_tree is taking over 10 seconds
2020-04-24 09:16:42 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.christmas_tree fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 476, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/meross_cloud/switch.py", line 34, in update
    self._device.get_status(force_status_refresh=True)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 134, in get_status
    self._state = self._get_status_impl()
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 88, in _get_status_impl
    data = self.get_sys_data()['all']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 119, in get_sys_data
    return self.execute_command("GET", ALL, {}, online_check=False)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 296, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10
2020-04-24 09:16:43 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform meross_cloud is taking over 10 seconds.
2020-04-24 09:16:52 WARNING (MainThread) [homeassistant.helpers.entity] Update of switch.outside_lights is taking over 10 seconds
2020-04-24 09:16:52 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up meross_cloud platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/config/custom_components/meross_cloud/sensor.py", line 175, in async_setup_entry
    sensor_entities = await hass.async_add_executor_job(sync_logic)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/meross_cloud/sensor.py", line 163, in sync_logic
    elif plug.type.startswith("mss310") or plug.supports_consumption_reading():
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 152, in supports_consumption_reading
    return CONSUMPTIONX in self.get_abilities()
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 124, in get_abilities
    self._abilities = self.execute_command("GET", ABILITY, {})['ability']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 296, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10
2020-04-24 09:16:52 ERROR (MainThread) [homeassistant.helpers.entity] Update for switch.outside_lights fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 279, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 476, in async_device_update
    await self.hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/meross_cloud/switch.py", line 34, in update
    self._device.get_status(force_status_refresh=True)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 134, in get_status
    self._state = self._get_status_impl()
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 88, in _get_status_impl
    data = self.get_sys_data()['all']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 119, in get_sys_data
    return self.execute_command("GET", ALL, {}, online_check=False)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 296, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10
2020-04-24 09:19:02 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1255, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    self._platforms.values(), func, call, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 412, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 600, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 443, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 625, in async_turn_on
    await self.hass.async_add_job(ft.partial(self.turn_on, **kwargs))
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/meross_cloud/switch.py", line 101, in turn_on
    self._device.turn_on_channel(self._channel_id)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 160, in turn_on_channel
    return self._channel_control_impl(c, 1, callback=callback)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/devices/power_plugs.py", line 27, in _channel_control_impl
    if TOGGLE in self.get_abilities():
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 124, in get_abilities
    self._abilities = self.execute_command("GET", ABILITY, {})['ability']
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/device.py", line 116, in execute_command
    return self.__cloud_client.execute_cmd(self.uuid, command, namespace, payload, callback=callback, timeout=timeout)
  File "/usr/local/lib/python3.7/site-packages/meross_iot/cloud/client.py", line 296, in execute_cmd
    raise CommandTimeoutException("A timeout occurred while waiting for the ACK: %d" % timeout)
meross_iot.cloud.exceptions.CommandTimeoutException.CommandTimeoutException: A timeout occurred while waiting for the ACK: 10

The log has been truncated but I can provide further details if required.

You can see from the snip below that the turn-on button appears to be unavailable.

![image](https://user-images.githubusercontent.com/53110372/80191628-dbe2eb80-860d-11ea-973f-c3ce2f507f42.png)

**Comment by Jonah1970** [comment]
Hi @albertogeniola, I have an update for you ... I saw there was an update 20200424.2 available so I upgraded to that at the situation was the same.  I then decided to reboot the 2 plugs in question and they then both started to work (I had not tried this prevously because they were working fine from the Meross App).  Following a susequent restart of HA the log is now "clean".
Thanks and I hope that helps you.
